### PR TITLE
feat!: use oclif/core v4

### PR DESCRIPTION
**Pull request by mdonnalley**
BREAKING CHANGE: remove table-related flags and standardize return value

- Use @oclif/core v4
- Remove the following flags
  - `--csv`
  - `--filter`
  - `--no-header`
  - `--output`

@W-15845231@

### CODE REVIEW

File: src/commands/commands.ts [hunk 1]
```diff
@@ -1,25 +1,104 @@
-import {Command, Flags, toConfiguredId, ux} from '@oclif/core'
+import {Command, Flags, toConfiguredId} from '@oclif/core'
 import _ from 'lodash'
-import {EOL} from 'node:os'
+// @ts-expect-error because object-treeify does not have types: https://github.com/blackflux/object-treeify/issues/1077
+import treeify from 'object-treeify'
+import TtyTable from 'tty-table'
+// import {EOL} from 'node:os'
```
Comment (iowillhoit) [inline src/commands/commands.ts:6]: Remove?

File: src/commands/commands.ts [hunk 2]
```diff
@@ -1,25 +1,104 @@
...
 
-import createCommandTree from '../utils/tree.js'
+// import createCommandTree from '../utils/tree.js'
```
Comment (iowillhoit) [inline src/commands/commands.ts:8]: Remove?

Comment (mdonnalley) [reply src/commands/commands.ts]: fixed - sorry for being sloppy!

**Comment by iowillhoit** [comment]
By default, the command name should not be truncated. Also, ideally the full command name (currently seeing with `--no-truncate`) should not wrap. 

Truncated:
Screenshot 2024-05-31 at 10 37 40 AM

Wrapping command names:
Screenshot 2024-05-31 at 10 40 39 AM

Before this PR:
Screenshot 2024-05-31 at 10 41 12 AM

**Comment by iowillhoit** [comment]
Much better! Looks good

**Review by iowillhoit — APPROVED** [review]

**Pull request merged**